src/nunit.analyzers.tests/Helpers/NUnitParamAttributeTypeConversionsTests.cs

```diff
@@ -0,0 +1,68 @@
+using Microsoft.CodeAnalysis;
+using NUnit.Analyzers.Helpers;
+using NUnit.Framework;
+
+namespace NUnit.Analyzers.Tests.Helpers
+{
+    [TestFixture]
+    internal sealed class NUnitParamAttributeTypeConversionsTests
+    {
+        [TestCase(SpecialType.System_Int16)]
+        [TestCase(SpecialType.System_Byte)]
+        [TestCase(SpecialType.System_SByte)]
+        [TestCase(SpecialType.System_Int64)]
+        [TestCase(SpecialType.System_Double)]
+        [TestCase(SpecialType.System_Decimal)]
+        public void CanConvertFromInt32ToOtherTypes(SpecialType targetType)
+        {
+            var valueType = SpecialType.System_Int32;
+            Assert.That(NUnitParamAttributeTypeConversions.CanConvert(valueType, targetType), Is.True);
+        }
+
+        [TestCase(SpecialType.System_Int32)]
+        [TestCase(SpecialType.System_Double)]
+        [TestCase(SpecialType.System_String)]
+        public void CanConvertFromOtherTypesToDecimal(SpecialType valueType)
+        {
+            var targetType = SpecialType.System_Decimal;
+            Assert.That(NUnitParamAttributeTypeConversions.CanConvert(valueType, targetType), Is.True);
+        }
+
+        [TestCase(SpecialType.System_String)]
+        public void CanConvertFromOtherTypesToDateTime(SpecialType valueType)
+        {
+            var targetType = SpecialType.System_DateTime;
+            Assert.That(NUnitParamAttributeTypeConversions.CanConvert(valueType, targetType), Is.True);
+        }
+
+        [TestCase(SpecialType.System_SByte)]
+        [TestCase(SpecialType.System_Byte)]
+        [TestCase(SpecialType.System_Int16)]
+        [TestCase(SpecialType.System_UInt16)]
+        [TestCase(SpecialType.System_Int32)]
+        [TestCase(SpecialType.System_UInt32)]
+        [TestCase(SpecialType.System_Int64)]
+        [TestCase(SpecialType.System_UInt64)]
+        [TestCase(SpecialType.System_Single)]
+        [TestCase(SpecialType.System_Double)]
+        [TestCase(SpecialType.System_Decimal)]
+        [TestCase(SpecialType.System_DateTime)]
+        public void CanConvertFromSameTypeToItself(SpecialType valueType)
+        {
+            var targetType = valueType;
+            Assert.That(NUnitParamAttributeTypeConversions.CanConvert(valueType, targetType), Is.True);
+        }
+
+        [TestCase(SpecialType.System_Int32, SpecialType.System_UInt16)]
+        [TestCase(SpecialType.System_Int32, SpecialType.System_UInt32)]
+        [TestCase(SpecialType.System_Int32, SpecialType.System_UInt64)]
+        [TestCase(SpecialType.System_UInt32, SpecialType.System_UInt16)]
+        [TestCase(SpecialType.System_UInt32, SpecialType.System_UInt64)]
```

Comment:
I was a little surprised that one cannot convert from `SpecialType.System_UInt32` to `SpecialType.System_UInt64`